crates/sui-metric-checker/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, NaiveDateTime, Utc};
+use humantime::parse_duration;
+use serde::Deserialize;
+use strum_macros::Display;
+
+pub mod query;
+
+#[derive(Debug, Display, Deserialize)]
+pub enum QueryType {
+    Instant,
+    Range,
+}
+
+#[derive(Debug, Display, Deserialize)]
+pub enum ErrorCondition {
+    Greater,
+    Equal,
+    Less,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct QueryResultValidation {
```

Comment:
It will be useful to have an example, e.g. threshold = 10, ErrorCondition = Greater, value = 5 vs 20, which succeeds and which fails?